device/tt_device.h

```diff
@@ -243,6 +243,7 @@ class tt_device
      *
      * @param host_address_params_ All the Host Address space parameters required by UMD.
      */ 
+    [[deprecated("Using unnecessary function.")]]
     virtual void set_driver_host_address_params(const tt_driver_host_address_params& host_address_params_) {
```

Comment:
There is some nuance here that isn't captured but probably should be, irrespective of this change.

UMD gives the application full control over the sysmem.  By calling this method, the application is acknowledging that it knows that part of sysmem is 'off limits' because UMD and ETH firmware are using it for remote IO.

I agree that the parameters being set here can and should be UMD implementation detail.  However, I don't see any documentation that describes the effect this method has or the implications for how the application uses sysmem.